### PR TITLE
[feat] add triton & flydsl impl for sparse  attention

### PR DESCRIPTION
**Pull request by kyle-256**
# Description

Add DeepSeek-V4 **sparse-MLA attention** (single-latent, top-k sparse) as native FlyDSL forward + backward kernels for gfx950 / MI355X, plus a plain-Triton reference implementation used as the correctness oracle. Covers the pure-SWA (cr=0), random-pool (cr=4) and deterministic-pool / HCA (cr=128) variants for both `flash` (H=64) and `pro` (H=128) head counts, with softmax-sink support and a deterministic (no-atomic) backward.

Fixes # (N/A)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [x] Code refactoring

## Changes

- **FlyDSL sparse-MLA kernels** (`primus_turbo/flydsl/attention/sparse_mla_{fwd,bwd}.py`): forward (flash softmax + sink) and backward (deterministic dQ / dKV-intermediate / gather split, no float atomics); dispatches per shape across cr in {0, 4, 128} x {flash, pro}.
- **Triton reference** (`primus_turbo/triton/attention/sparse_mla.py`): plain-Triton fwd + bwd, single file, used as the SNR oracle.
- **Shared helpers**: `gemm_helper.make_bf16_rebased_rsrc` (>4 GB rebased bf16 buffer SRD, used by bwd); `triton_knobs_helper.scoped_amd_triton_knobs_disabled` (scoped ping-pong / async-copy disable for the sparse-MLA kernels — the global GEMM knobs pessimize them and overflow LDS on the wide dKV tiling).
- **Benchmark** (`benchmark/ops/training/bench_sparse_mla.py`): TFLOPS + SNR-vs-triton over the 6 shapes, styled after `bench_attention_turbo.py`.
- **Tests** (`tests/pytorch/ops/test_attention.py::test_sparse_mla`): fwd/bwd SNR vs the triton oracle (> 40 dB) + bitwise determinism, over 6 shapes x seqlen {512, 1024, 2048} (512 also exercises the cr=4 small-seq dkv-dispatch guard); gfx950-only skip.
- **Quality**: no env knobs (all baked in), comments <= 5 lines with no benchmark numbers, descriptive names (no `v4` / debug-style suffixes), ruff clean.

# Checklist:

- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: benchmark/ops/training/bench_sparse_mla.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
+###############################################################################
```
Comment (RuibinCheung) [inline benchmark/ops/training/bench_sparse_mla.py]: Unify it into https://github.com/AMD-AGI/Primus-Turbo/blob/main/benchmark/ops/training/bench_attention_turbo.py when function is ready.

**Review by RuibinCheung — CHANGES_REQUESTED** [review]

**Review by RuibinCheung — APPROVED** [review]
LGTM if CI is green.
